Lib/Src/CThermistor.cpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file CThermistor.cpp
+ */
+
+/*
+ *  Created on: Feb 15, 2022
+ *      Author: niuslar
+ */
+
+#include "CThermistor.h"
+#include "math.h"
+
+#define MIN_VOLT_RANGE     (0.57)
+#define MAX_VOLT_RANGE     (2.67)
+#define OUT_OF_RANGE_VALUE (999.9)
+
+CUartCom uart_temp(&huart2);
+CLog log_temp(&uart_temp, "Temp_sensor");
+
+/**
+ * @brief Constructor
+ * @param adc_data ADC Data object
+ * @param adc_channel starting from 0
+ */
+CThermistor::CThermistor(CAdcData &adc_data, uint8_t adc_channel)
+    : m_adc_data(adc_data),
+      m_adc_channel(adc_channel)
+{
+    // Check channel
+    if (m_adc_channel >= ADC_CHANNELS)
+    {
+        // Log warning
+        log_temp.log(CLog::LOG_WARNING, "Invalid ADC Channel");
+    }
+}
+
+float CThermistor::getTemperature()
```

Comment:
```
float CThermistor::getTemperature(float voltage)
{
    float temperature = 0;
    // here you multiply voltage by the calibration coefficients stored in the thermistor.
    return temperature;
}

Reply:
Do you mean something like: 
```
 temp_celsius = m_coeff_array[0] - m_coeff_array[1] * voltage + m_coeff_array[2] * pow(voltage, 2) - m_coeff_array[3] * pow(voltage, 3);
```
?

Reply:
ah. I forgot about that.
Nope. Let me do a branch and show you.